### PR TITLE
[CARBONDATA-903] data load is not failing even though bad records exists in the data in case of unsafe sort or batch sort

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1570/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1572/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1573/

**Comment by ravipesala** [comment]
@mohammadshahidkhan Please create a Abstract class and move the common methods to there.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1580/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1582/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1584/

File: processing/src/main/java/org/apache/carbondata/processing/newflow/sort/AbstractMergeSorter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+package org.apache.carbondata.processing.newflow.sort;
+
+import org.apache.carbondata.processing.newflow.exception.CarbonDataLoadingException;
+import org.apache.carbondata.processing.newflow.sort.impl.ThreadStatusObserver;
+
+/**
+ * The class defines the common methods used in across various type of sort
+ */
+public abstract class AbstractMergeSorter {
```
Comment (ravipesala) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/sort/AbstractMergeSorter.java]: implement it with sorter

Comment (mohammadshahidkhan) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/sort/AbstractMergeSorter.java]: fixed

File: processing/src/main/java/org/apache/carbondata/processing/newflow/sort/impl/UnsafeBatchParallelReadMergeSorterImpl.java [hunk 2]
```diff
@@ -197,7 +209,12 @@ private void createSortDataRows() {
 
     @Override public UnsafeSingleThreadFinalSortFilesMerger next() {
       try {
-        return mergerQueue.take();
+        UnsafeSingleThreadFinalSortFilesMerger unsafeSingleThreadFinalSortFilesMerger =
```
Comment (mohammadshahidkhan) [inline processing/src/main/java/org/apache/carbondata/processing/newflow/sort/impl/UnsafeBatchParallelReadMergeSorterImpl.java]: @ravipesala  Please review portion 
Here i have fixed the dead lock issue
https://issues.apache.org/jira/browse/CARBONDATA-913

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1594/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**